dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -307,17 +321,45 @@ public String getActionIdByName(final String actionName,
                                     final Contentlet contentlet,
                                     final User user) throws DotSecurityException, DotDataException {
 
-        final List<WorkflowAction> availableActionsOnListing =
-                APILocator.getWorkflowAPI().findAvailableActionsListing(contentlet, user);
+        final WorkflowAPI workflowAPI                = APILocator.getWorkflowAPI();
+        final Optional<WorkflowStep> workflowStepOpt = workflowAPI.findCurrentStep(contentlet);
+        Optional<WorkflowScheme> schemeOpt           = Optional.empty();
+        Optional<WorkflowAction> foundAction         = Optional.empty();
+        if (workflowStepOpt.isPresent()) {
+            // 1) look for the actions on the same step
+            final WorkflowStep workflowStep = workflowStepOpt.get();
+            foundAction = this.findActionsOn(actionName, workflowAPI.findActions(workflowStep, user));
+            if (foundAction.isPresent()) {
+
+                return foundAction.get().getId();
+            }
 
-        final List<WorkflowAction> availableActionsOnEditing =
-                APILocator.getWorkflowAPI().findAvailableActionsEditing(contentlet, user);
+            // 2) look for the actions on the current scheme
+            final WorkflowScheme workflowScheme = workflowAPI.findScheme(workflowStep.getSchemeId());
+            schemeOpt   = Optional.ofNullable(workflowScheme);
+            foundAction = this.findActionsOn(actionName, workflowAPI.findActions(workflowScheme, user));
+            if (foundAction.isPresent()) {
 
-        final Optional<WorkflowAction> foundAction =
-                Stream.concat(availableActionsOnListing.stream(), availableActionsOnEditing.stream())
-                        .filter(action -> action.getName().equalsIgnoreCase(actionName)).findFirst();
+                return foundAction.get().getId();
+            }
+        }
 
-        return foundAction.isPresent()?foundAction.get().getId():null;
+        // 3) look for the actions on all schemes.
+        final List<WorkflowScheme>  workflowSchemes = workflowAPI.findSchemesForContentType(contentlet.getContentType());
+        for (final WorkflowScheme scheme : workflowSchemes) {
+
+            if (schemeOpt.isPresent() && schemeOpt.get().getId().equals(scheme.getId()))  {
+                continue; // we already analized this scheme
+            }
+
+            foundAction = this.findActionsOn(actionName, workflowAPI.findActions(scheme, user));
+            if (foundAction.isPresent()) {
+
+                return foundAction.get().getId();
+            }
+        }
+
+        return null;
```

Comment:
maybe this method should return a Optional

Reply:
yes, but not than bad